cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/ApplicationLifecycleServiceTest.java

```diff
@@ -152,6 +154,42 @@ public void testOwner() throws Exception {
   }
 
   @Test
+  public void testOwnerInImpersonatedNS() throws Exception {
```

Comment:
Add a test without a namespace owner? App impersonation should work in a namespace without an explicitly set principal too.